packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -1,5 +1,88 @@
-function reducer(state = {}): typeof state {
-  return state;
+import { AnyAction, Dispatch } from 'redux';
+import { getAggregations } from './../utlis/aggregations';
+import { getQueries } from './../utlis/queries';
+
+enum actions {
+  ITEMS_FETCHED = 'itemsFetched',
 }
 
+type StateActions = {
+  type: actions.ITEMS_FETCHED;
+  payload: Item[];
+};
+
+export type Item = {
+  id: string;
+  lastModified: number;
+  name: string;
+  namespace: string;
+  type: 'query' | 'aggregation';
+};
+
+export type State = {
+  loading: boolean;
+  items: Item[];
+};
+
+const INITIAL_STATE: State = {
+  loading: true,
+  items: [],
+};
+
+function reducer(
+  state = INITIAL_STATE,
+  action: StateActions | AnyAction
+): State {
+  const newState = { ...state };
+  if (action.type === actions.ITEMS_FETCHED) {
+    newState.items = action.payload;
+    newState.loading = false;
+  }
+  return newState;
+}
+
+export const fetchItems = () => {
+  return async (dispatch: Dispatch<StateActions>): Promise<void> => {
+    let aggregations: Item[] = [];
+    let queries: Item[] = [];
+    try {
+      aggregations = await getAggregationItems();
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+    try {
+      queries = await getQueryItems();
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    const payload = [...aggregations, ...queries];
+    payload.sort((a, b) => a.lastModified - b.lastModified);
```

Comment:
I'd probably suggest not doing any sort here, we will need to implement it in the react view for the sort filter anyway and it's a bit better I think if all the sorting happens in one place

```suggestion
```